components/utilities/ulog/ulog.c

```diff
@@ -318,9 +318,12 @@ RT_WEAK rt_size_t ulog_formater(char *log_buf, rt_uint32_t level, const char *ta
         /* is not in interrupt context */
         if (rt_interrupt_get_nest() == 0)
         {
-            rt_size_t name_len = rt_strnlen(rt_thread_self()->name, RT_NAME_MAX);
-
-            rt_strncpy(log_buf + log_len, rt_thread_self()->name, name_len);
+            rt_size_t name_len = 0;
+            const char *thread_name = "N/A";
+            if (rt_thread_self()->name)
```

Comment:
or
```c
if (rt_thread_self())
{
    // ...
}
```

Reply:
Done

Reply:
Done? No. If the scheduler is not started, the rt_thread_self() returns `RT_NULL`. Then the `rt_thread_self()->name` will be the text code possiblely or hardfault in some machine.